[SUPERSEDED] Remove adaptation of 0-arg methods under -Xsource:3.0 and preserve infix unit value

som-snytt opened the pull request:
Commits not reordered or squashed.

Besides updating check files, attempts to preserve unit value in

```
"" + ()
```
which currently warns because it takes the empty parens as an empty arg list instead of unit value.

The spec is clear that "several args in parens" are taken as an arg list, not empty parens.

eed3si9n committed:
Improve the tuple adaptation warning message

Get thee to a nullary, go. Farewell -Xfuture.

som-snytt committed:
Retain empty parens as unit value

When parsing infix expression, a Parens node
is created for parens. If it is empty, they
wrote unit value. Otherwise, per the spec,
the possibly many comma-separated arguments
are taken as arguments.

`() == ()` is now what it is.

`x op()` is corrected to `x.op()`. The dot-averse
must rely on postfix and empty application
to use `x op`.

Restore wording adapt is removed.

som-snytt commented:
Hey, who put that red ex there?

som-snytt commented:
This looked like something weird, but actually it's simple:

`tree GETCLASS()` is bad syntax, and now we notice.

SethTisue commented:
needs rebase

som-snytt commented:
Thanks @SethTisue . I don't even care if you are @SethTisue or @SethTisueBot. I appreciate how you get after me, the way the Roomba doesn't actually clean the room, but reminds me to get the vacuum cleaner out of the utility closet and apply it. In the sense of "appliance". My efforts may be soon forgotten, or swept under the rug, to extend the previous metaphor, but yours will live on in infamy, or maybe infamily, to coin what may be a really important word someday.

SethTisue commented:
quoth the raven: needs rebase

adriaanm commented:
Sorry, I forgot about this one. I was looking at what remained TODO in https://github.com/scala/scala-dev/issues/471, and this PR was not referenced.

How about focussing this PR on the last commit? #7328, I think, subsumes the first two (it's a little less ambitious/polished, but it will do, I'd say)

SethTisue commented:
note that #7328 is now merged

som-snytt commented:
I cherry-picked the first commit and junit passed. Maybe I'll rebuild?

som-snytt commented:
/rebuild

som-snytt commented:
Rebased and squashed for lrytz's :eyes: .

Also TIL :eye_speech_bubble: for [anti-bullying](https://emojipedia.org/eye-in-speech-bubble/), which is another reason lrytz is the fellow to have on your PR review board.

The M$ icon looks more like "big brother is watching everything you say."

som-snytt commented:
/synch

som-snytt commented:
Oh thanks @adriaanm I see you recently took the review upon yourself. You needn't feel too bad that Eugene's change has been waiting since last June: "get thee to a nullary" is still funny!

Rebasing is getting kind of stale, however.

SethTisue commented:
superseded by #7684
